docs/spec.md

```diff
@@ -149,74 +142,76 @@ to that of the derivation (or derivations).
 * [Traversable](#traversable)
 
 
+### Setoid
 
-## Setoid
-
-#### Methods
-
-  1. `equals :: Setoid s => Type s ~> (s, s) → Boolean`
+```js
+Setoid<T> {
+  equals: (T, T) => boolean
+}
+```
 
-#### Laws
+Module must match the `Setoid` signature for some type `T`, and obey following laws:
 
   1. Reflexivity: `S.equals(a, a) === true`
   1. Symmetry: `S.equals(a, b) === S.equals(b, a)`
   1. Transitivity: if `S.equals(a, b)` and `S.equals(b, c)`, then `S.equals(a, c)`
 
 
+### Semigroup
 
-## Semigroup
-
-#### Methods
-
-  1. `concat :: Semigroup s => Type s ~> (s, s) → s`
+```js
+Semigroup<T> {
+  concat: (T, T) => T
+}
+```
 
-#### Laws
+Module must match the `Semigroup` signature for some type `T`, and obey following laws:
 
   1. Associativity: `S.concat(S.concat(a, b), c) ≡ S.concat(a, S.concat(b, c))`
 
 
+### Monoid
 
-## Monoid
-
-#### Dependencies
-
-  1. Semigroup
-
-#### Methods
-
-  1. `empty :: Monoid m => Type m ~> () → m`
+```js
+Monoid<T> {
+  empty: () => T
+}
+```
 
-#### Laws
+Module must match the `Monoid` signature for some type `T`,
+support `Semigroup` algebra for the same `T`, and obey following laws:
 
   1. Right identity: `M.concat(a, M.empty()) ≡ a`
   1. Left identity: `M.concat(M.empty(), a) ≡ a`
 
 
+### Functor
 
-## Functor
-
-#### Methods
-
-  1. `map :: Functor f => Type f ~> (a → b, f a) → f b`
+```js
+Functor<T> {
+  map: (a => b, T<a>) => T<b>
+}
+```
 
-#### Laws
+Module must match the `Functor` signature for some type `T`, and obey following laws:
 
   1. Identity: `F.map(x => x, a) ≡ a`
   1. Composition: `F.map(x => f(g(x)), a) ≡ F.map(f, F.map(g, a))`
 
 
 
-## Bifunctor
-
-#### Dependencies
+### Bifunctor
 
-  1. Functor
-
-#### Methods
-
-  1. `bimap :: Bifunctor f => Type f ~> (a → b, c → d, f a c) → f b d`
+```js
+Bifunctor<T> {
+  bimap: (a => b, c => d, T<a, c>) => T<b, d>
+}
+```
```

Comment:
I think a english sentence does better job of explaining that tricky relationship we have with Bifunctor and Functor. If you use syntax that you suggest I feel like we'll still have to complement it with an explanation in english, so why use syntax at all in this case?